spec/manual/find_executable_spec.cr

```diff
@@ -0,0 +1,65 @@
+# Verifies that find_executable's specs match the behavior of Process.new
+# This doesn't actually test find_executable, only takes all the test cases
+# directly from spec/std/process/find_executable_spec.cr and checks that
+# *they* match what the OS actually does when finding an executable for the
+# purpose of running it.
+
+require "spec"
+require "digest/sha1"
+require "../support/tempfile"
+require "../std/process/find_executable_spec"
+
+describe "Process.run" do
+  test_dir = Path[SPEC_TEMPFILE_PATH] / "manual_find_executable"
+  base_dir = Path[test_dir] / "base"
+  path_dir = Path[test_dir] / "path"
+
+  around_all do |all|
+    Dir.mkdir_p(test_dir)
+
+    exe_names, file_names = find_executable_test_files
+
+    exe_names.map do |name|
+      src = "print #{name.inspect}"
+      digest = Digest::SHA1.hexdigest(src)
+      src_fn = test_dir / "#{digest}.cr"
+      exe_fn = test_dir / "#{digest}.exe"
+      File.write(src_fn, src)
+      {name, exe_fn, Process.new("crystal", ["build", "-o", exe_fn.to_s, src_fn.to_s])}
```

Comment:
Spawning a dozen parallel compiler instances seems a bit heavy even for a manual spec.

Reply:
Well, didn't want to make it too complicated or too slow. Suggestions?